After new release milestone 8 building process fails in testing section.

# Bug Report
After new release milestone 8 building process fails in testing section.

## Describe the Bug
After new release milestone 8 building process fails in testing section with following trace:
```
$ ./gradlew build -x test;
> Task :system-tests:compileTestJava
/home/aramirez/PROJECTS/aramirez/REPOSITORIES/EDC-MinimumViableDataspace/system-tests/src/test/java/org/eclipse/edc/system/tests/identityhub/IdentityHubIntegrationTest.java:135: error: cannot find symbol
            var jwt = ((JwtCredentialEnvelope) envelope).getJwtVerifiableCredentials();
                                                        ^
  symbol:   method getJwtVerifiableCredentials()
  location: class JwtCredentialEnvelope
```

### Expected Behavior
Building process running correctly.

### Observed Behavior
It fails on building process

## Steps to Reproduce
1. Execute in command line the following command as it is told in documentation:
   https://github.com/eclipse-edc/MinimumViableDataspace/blob/main/system-tests/README.md

## Context Information
Nothing to add.

## Detailed Description
As commented before.

## Possible Implementation
Adapt current MVD code to 0.0.1-milestone-8 release for following repositories:
- https://github.com/eclipse-edc/Connector/releases/tag/v0.0.1-milestone-8
- https://github.com/eclipse-edc/GradlePlugins/releases/tag/v0.0.1-milestone-8
- https://github.com/eclipse-edc/IdentityHub/releases/tag/v0.0.1-milestone-8
- https://github.com/eclipse-edc/RegistrationService/releases/tag/v0.0.1-milestone-8
- https://github.com/eclipse-edc/FederatedCatalog/releases/tag/v0.0.1-milestone-8

--- a/deployment/azure/seed_vcs.sh
+++ b/deployment/azure/seed_vcs.sh
@@ -17,15 +17,15 @@ asset_account="$7"
 ## Function declarations to be used later
 pushCredential() {
   local participant="$1"
-  local credential="$2"
-  echo "Push credentials to $participant at port $identityPort"
-  echo "    credential: ${credential}"
+  local claims="$2"
+  echo "Push claims to $participant at port $identityPort"
+  echo "    claims: ${claims}"
   echo
   local participant_did="did:web:$participant_did_host"
   local gaiax_did="$3"
 
   java -jar identity-hub-cli.jar -s="$ihUrl" vc add \
-    -c="$credential" \
+    -c="$claims" \
     -b="$participant_did" \
     -i="$gaiax_did" \
     -k="terraform/generated/dataspace/gaiaxkey.pem"
@@ -62,9 +62,6 @@ echo
 
 # hack - assume all containers have sequential management api dataPort configurations, check docker/docker-compose.yml for details!!!
 
-id=$(uuidgen)
-pushCredential "$participant" '{"id": "'$id'", "credentialSubject": {"gaiaXMember": "true"}}' "$gaiax_did"
-id=$(uuidgen)
-pushCredential "$participant" '{"id": "'$id'", "credentialSubject": {"region": "'$region'"}}' "$gaiax_did"
+pushCredential "$participant" '{"region": "'"$region"'"}' "$gaiax_did"
 
 checkCredentials

--- a/extensions/policies/build.gradle.kts
+++ b/extensions/policies/build.gradle.kts
@@ -20,7 +20,6 @@ dependencies {
     api(edc.spi.ids)
     api(edc.spi.contract)
     api(edc.core.connector)
-    api(identityHub.ext.verifier.jwt)
     implementation(identityHub.spi.core)
 
     testImplementation(edc.policy.engine)

--- a/extensions/policies/src/main/java/org/eclipse/edc/mvd/RegionConstraintFunction.java
+++ b/extensions/policies/src/main/java/org/eclipse/edc/mvd/RegionConstraintFunction.java
@@ -14,8 +14,7 @@
 
 package org.eclipse.edc.mvd;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import org.eclipse.edc.identityhub.spi.credentials.model.VerifiableCredential;
+import org.eclipse.edc.identityhub.spi.credentials.model.Credential;
 import org.eclipse.edc.policy.engine.spi.AtomicConstraintFunction;
 import org.eclipse.edc.policy.engine.spi.PolicyContext;
 import org.eclipse.edc.policy.model.Operator;
@@ -28,15 +27,12 @@ import java.util.Collections;
 import java.util.List;
 import java.util.Map;
 
-import static org.eclipse.edc.identityhub.verifier.jwt.VerifiableCredentialsJwtService.VERIFIABLE_CREDENTIALS_KEY;
-
 public class RegionConstraintFunction implements AtomicConstraintFunction<Permission> {
+    
     private static final String REGION_KEY = "region";
-    private final ObjectMapper objectMapper;
     private final Monitor monitor;
 
-    public RegionConstraintFunction(ObjectMapper objectMapper, Monitor monitor) {
-        this.objectMapper = objectMapper;
+    public RegionConstraintFunction(Monitor monitor) {
         this.monitor = monitor;
     }
 
@@ -70,9 +66,8 @@ public class RegionConstraintFunction implements AtomicConstraintFunction<Permis
     }
 
     private String getRegion(Object object) {
-        var vcObject = (Map<String, Object>) object;
-        var verifiableCredentialMap = vcObject.get(VERIFIABLE_CREDENTIALS_KEY);
-        var verifiableCredential = objectMapper.convertValue(verifiableCredentialMap, VerifiableCredential.class);
-        return (String) (verifiableCredential.getCredentialSubject().get(REGION_KEY));
+        var credential = (Credential) object;
+        var claims = credential.getCredentialSubject().getClaims();
+        return (String) claims.get(REGION_KEY);
     }
 }

--- a/extensions/policies/src/main/java/org/eclipse/edc/mvd/SeedPoliciesExtension.java
+++ b/extensions/policies/src/main/java/org/eclipse/edc/mvd/SeedPoliciesExtension.java
@@ -66,7 +66,7 @@ public class SeedPoliciesExtension implements ServiceExtension {
         ruleBindingRegistry.bind("USE", ALL_SCOPES);
         ruleBindingRegistry.bind(ABS_SPATIAL_POSITION, ContractDefinitionService.CATALOGING_SCOPE);
 
-        policyEngine.registerFunction(ALL_SCOPES, Permission.class, ABS_SPATIAL_POSITION, new RegionConstraintFunction(typeManager.getMapper(), monitor));
+        policyEngine.registerFunction(ALL_SCOPES, Permission.class, ABS_SPATIAL_POSITION, new RegionConstraintFunction(monitor));
     }
 
 }

--- a/extensions/policies/src/test/java/org/eclipse/edc/mvd/RegionConstraintFunctionTest.java
+++ b/extensions/policies/src/test/java/org/eclipse/edc/mvd/RegionConstraintFunctionTest.java
@@ -14,44 +14,44 @@
 
 package org.eclipse.edc.mvd;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
+import org.eclipse.edc.identityhub.spi.credentials.model.Credential;
+import org.eclipse.edc.identityhub.spi.credentials.model.CredentialSubject;
 import org.eclipse.edc.policy.engine.PolicyContextImpl;
 import org.eclipse.edc.policy.engine.spi.PolicyContext;
 import org.eclipse.edc.policy.model.Operator;
 import org.eclipse.edc.policy.model.Permission;
 import org.eclipse.edc.spi.agent.ParticipantAgent;
-import org.eclipse.edc.spi.monitor.ConsoleMonitor;
 import org.eclipse.edc.spi.monitor.Monitor;
 import org.junit.jupiter.api.Test;
 
+import java.sql.Date;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
 import java.util.List;
 import java.util.Map;
 import java.util.UUID;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.eclipse.edc.identityhub.verifier.jwt.VerifiableCredentialsJwtService.VERIFIABLE_CREDENTIALS_KEY;
+import static org.mockito.Mockito.mock;
 
 class RegionConstraintFunctionTest {
-    private static final String VERIFIABLE_CREDENTIAL_ID_KEY = "id";
-    private static final String CREDENTIAL_SUBJECT_KEY = "credentialSubject";
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-    private static final Monitor MONITOR = new ConsoleMonitor();
-    private static final RegionConstraintFunction CONSTRAINT_FUNCTION = new RegionConstraintFunction(OBJECT_MAPPER, MONITOR);
+
+    private static final Monitor MONITOR = mock(Monitor.class);
+    private static final RegionConstraintFunction CONSTRAINT_FUNCTION = new RegionConstraintFunction(MONITOR);
     private static final Permission PERMISSION = Permission.Builder.newInstance().build();
     private static final String REGION_KEY = "region";
     private static final String EXPECTED_REGION = "eu";
-    private static final String ISSUER_KEY = "iss";
 
     @Test
     void verifyPolicy_validRegion() {
-        var claims = toMappedVerifiableCredentials(Map.of(REGION_KEY, EXPECTED_REGION));
+        var claims = toMappedVerifiableCredentials(REGION_KEY, EXPECTED_REGION);
         var policyContext = getPolicyContext(claims);
         assertThat(CONSTRAINT_FUNCTION.evaluate(Operator.EQ, EXPECTED_REGION, PERMISSION, policyContext)).isTrue();
     }
 
     @Test
     void verifyPolicy_invalidRegion() {
-        var claims = toMappedVerifiableCredentials(Map.of(REGION_KEY, "us"));
+        var claims = toMappedVerifiableCredentials(REGION_KEY, "us");
         var policyContext = getPolicyContext(claims);
         assertThat(CONSTRAINT_FUNCTION.evaluate(Operator.EQ, EXPECTED_REGION, PERMISSION, policyContext)).isFalse();
     }
@@ -66,42 +66,42 @@ class RegionConstraintFunctionTest {
     @Test
     void verifyPolicy_invalidRegionFormat() {
         // Region is a map instead of a string.
-        var claims = toMappedVerifiableCredentials(Map.of(REGION_KEY, Map.of()));
+        var claims = toMappedVerifiableCredentials(REGION_KEY, Map.of());
         var policyContext = getPolicyContext(claims);
         assertThat(CONSTRAINT_FUNCTION.evaluate(Operator.EQ, EXPECTED_REGION, PERMISSION, policyContext)).isFalse();
     }
 
     @Test
     void verifyPolicy_unsupportedOperator() {
-        var claims = toMappedVerifiableCredentials(Map.of(REGION_KEY, EXPECTED_REGION));
+        var claims = toMappedVerifiableCredentials(REGION_KEY, EXPECTED_REGION);
         var policyContext = getPolicyContext(claims);
         assertThat(CONSTRAINT_FUNCTION.evaluate(Operator.GT, EXPECTED_REGION, PERMISSION, policyContext)).isFalse();
     }
 
     @Test
     void verifyPolicy_NeqOperatorValidRegion() {
-        var claims = toMappedVerifiableCredentials(Map.of(REGION_KEY, "us"));
+        var claims = toMappedVerifiableCredentials(REGION_KEY, "us");
         var policyContext = getPolicyContext(claims);
         assertThat(CONSTRAINT_FUNCTION.evaluate(Operator.NEQ, EXPECTED_REGION, PERMISSION, policyContext)).isTrue();
     }
 
     @Test
     void verifyPolicy_NeqOperatorInvalidRegion() {
-        var claims = toMappedVerifiableCredentials(Map.of(REGION_KEY, EXPECTED_REGION));
+        var claims = toMappedVerifiableCredentials(REGION_KEY, EXPECTED_REGION);
         var policyContext = getPolicyContext(claims);
         assertThat(CONSTRAINT_FUNCTION.evaluate(Operator.NEQ, EXPECTED_REGION, PERMISSION, policyContext)).isFalse();
     }
 
     @Test
     void verifyPolicy_InOperatorValidRegion() {
-        var claims = toMappedVerifiableCredentials(Map.of(REGION_KEY, EXPECTED_REGION));
+        var claims = toMappedVerifiableCredentials(REGION_KEY, EXPECTED_REGION);
         var policyContext = getPolicyContext(claims);
         assertThat(CONSTRAINT_FUNCTION.evaluate(Operator.IN, List.of(EXPECTED_REGION), PERMISSION, policyContext)).isTrue();
     }
 
     @Test
     void verifyPolicy_InOperatorInValidRegion() {
-        var claims = toMappedVerifiableCredentials(Map.of(REGION_KEY, "us"));
+        var claims = toMappedVerifiableCredentials(REGION_KEY, "us");
         var policyContext = getPolicyContext(claims);
         assertThat(CONSTRAINT_FUNCTION.evaluate(Operator.IN, List.of(EXPECTED_REGION), PERMISSION, policyContext)).isFalse();
     }
@@ -110,12 +110,19 @@ class RegionConstraintFunctionTest {
         return new PolicyContextImpl(new ParticipantAgent(claims, Map.of()), Map.of());
     }
 
-    private Map<String, Object> toMappedVerifiableCredentials(Map<String, Object> regionClaims) {
-        var vcId = UUID.randomUUID().toString();
-        return Map.of(vcId,
-                Map.of(VERIFIABLE_CREDENTIALS_KEY, Map.of(CREDENTIAL_SUBJECT_KEY, regionClaims,
-                                VERIFIABLE_CREDENTIAL_ID_KEY, vcId),
-                        // issuer will be ignored when applying policies for now.
-                        ISSUER_KEY, String.join("did:web:", UUID.randomUUID().toString())));
+    private Map<String, Object> toMappedVerifiableCredentials(String key, Object value) {
+        var credentialId = UUID.randomUUID().toString();
+        var credential = Credential.Builder.newInstance()
+                .id("test")
+                .context("test")
+                .type("VerifiableCredential")
+                .issuer("did:web:" + UUID.randomUUID())
+                .issuanceDate(Date.from(Instant.now().truncatedTo(ChronoUnit.SECONDS)))
+                .credentialSubject(CredentialSubject.Builder.newInstance()
+                        .id("test")
+                        .claim(key, value)
+                        .build())
+                .build();
+        return Map.of(credentialId, credential);
     }
 }

--- a/launchers/connector/build.gradle.kts
+++ b/launchers/connector/build.gradle.kts
@@ -29,8 +29,7 @@ dependencies {
     implementation(edc.api.dataManagement)
     implementation(edc.config.filesystem)
     implementation(edc.ext.http)
-
-
+    
     // IDS
     implementation(edc.ids) {
         // Workaround for https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/issues/1387
@@ -65,8 +64,10 @@ dependencies {
     // Identity Hub
     implementation(identityHub.core)
     implementation(identityHub.ext.api)
+    implementation(identityHub.ext.selfdescription.api)
     implementation(identityHub.core.verifier)
     implementation(identityHub.ext.credentials.jwt)
+    implementation(identityHub.ext.verifier.jwt)
 }
 
 application {

--- a/launchers/registrationservice/build.gradle.kts
+++ b/launchers/registrationservice/build.gradle.kts
@@ -20,8 +20,8 @@ plugins {
 
 dependencies {
     implementation(registrationService.core)
-    implementation(registrationService.ext.participantVerifier)
-    implementation(registrationService.ext.registrationPolicy.gaiaxMember)
+    implementation(registrationService.core.credential.service)
+    implementation(registrationService.ext.api)
 
     implementation(edc.identity.did.web)
     implementation(edc.identity.did.core)
@@ -34,8 +34,10 @@ dependencies {
     runtimeOnly(edc.micrometer.jersey)
     implementation(edc.config.filesystem)
     implementation(identityHub.ext.api)
+    implementation(identityHub.ext.selfdescription.api)
     implementation(identityHub.core.verifier)
     implementation(identityHub.ext.credentials.jwt)
+    implementation(identityHub.ext.verifier.jwt)
 
     // To use FileSystem vault e.g. -DuseFsVault="true".Only for non-production usages.
     val useFsVault: Boolean = System.getProperty("useFsVault", "false").toBoolean()

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -35,6 +35,7 @@ dependencyResolutionManagement {
             ).versionRef("identityHub")
 
             library("ext-api", "org.eclipse.edc", "identity-hub-api").versionRef("identityHub")
+            library("ext-selfdescription-api", "org.eclipse.edc", "self-description-api").versionRef("identityHub")
             library(
                 "ext-verifier-jwt", "org.eclipse.edc", "identity-hub-verifier-jwt"
             ).versionRef("identityHub")
@@ -46,19 +47,13 @@ dependencyResolutionManagement {
         create("registrationService") {
             version("registrationService", "0.0.1-SNAPSHOT")
             library("core", "org.eclipse.edc", "registration-service").versionRef("registrationService")
+            library(
+                "core-credential-service",
+                "org.eclipse.edc",
+                "registration-service-credential-service"
+            ).versionRef("registrationService")
             library("core-client", "org.eclipse.edc", "registration-service-client").versionRef("registrationService")
-            library(
-                "ext-participantVerifier",
-                "org.eclipse.edc",
-                "participant-verifier"
-            ).versionRef("registrationService")
-
-            library(
-                "ext-registrationPolicy-gaiaxMember",
-                "org.eclipse.edc",
-                "registration-policy-gaiax-member"
-            ).versionRef("registrationService")
-
+            library("ext-api", "org.eclipse.edc", "registration-service-api").versionRef("registrationService")
         }
 
         create("fcc") {
@@ -66,7 +61,6 @@ dependencyResolutionManagement {
             library("api", "org.eclipse.edc", "federated-catalog-api").versionRef("catalog")
             library("spi", "org.eclipse.edc", "federated-catalog-spi").versionRef("catalog")
             library("core", "org.eclipse.edc", "federated-catalog-core").versionRef("catalog")
-
         }
 
         create("edc") {

--- a/system-tests/resources/cli-tools/entrypoint.sh
+++ b/system-tests/resources/cli-tools/entrypoint.sh
@@ -21,15 +21,14 @@ function seedVerifiedCredentials() {
   local participantDid="$2"
   local region="$3"
 
-   for subject in '"region": "'$region'"' '"gaiaXMember": "true"'
+   for subject in '"region": "'$region'"'
    do
      echo "Seeding VC for $participantName: $subject"
-     vcId=$(uuidgen)
      java -jar identity-hub-cli.jar \
                  -s="http://$participantName:7171/api/v1/identity/identity-hub" \
                  vc add \
-                 -c='{"id":"'"$vcId"'","credentialSubject":{'"$subject"'}}' \
-                -b="$participantDid" \
+                 -c="{ $subject }" \
+                 -b="$participantDid" \
                  -i="did:web:did-server:gaia-x" \
                  -k="/resources/vault/gaia-x/private-key.pem"
    done

--- a/system-tests/resources/vault/company1/company1-vault.properties
+++ b/system-tests/resources/vault/company1/company1-vault.properties
@@ -1,5 +1,8 @@
-#Wed Dec 07 13:12:22 UTC 2022
+#Wed Jan 25 15:30:42 UTC 2023
 b77fb664-30d7-4351-9c16-22f1ce71d2b0-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2022-12-07T14%3A10%3A32Z&sr\=c&sp\=w&sig\=x8uJEpVcSsk2JmNbjHyTBLe02Dy3aUeB8Y4pagDasZA%3D","expiration"\:1670422232589}
+7719e0bc-48a7-493f-a5a1-2b029252a891-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2023-01-25T16%3A30%3A40Z&sr\=c&sp\=w&sig\=7SsnR4c%2FjdXpl0n2IrHOg1fhyqypTPBlGhayZWnL3ao%3D","expiration"\:1674664240250}
+711c9e7c-6a94-4b6d-8f7a-36cf71cbc5bf-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2023-01-25T16%3A14%3A26Z&sr\=c&sp\=w&sig\=b%2FpbL5OmIx20MRA%2FMI1OtBOTuUu%2FSP4JTSxrr2piSQ0%3D","expiration"\:1674663266839}
+c42df689-100c-4111-b1e0-1af87c09e75a-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2022-12-07T14%3A07%3A18Z&sr\=c&sp\=w&sig\=FQqLkCUDQNtpJziMXtz6QzKlMlMtOCcjQKfF9CpgaBw%3D","expiration"\:1670422038984}
+2011fb1c-8fb8-497a-a687-418c2f2e5996-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2023-01-24T08%3A46%3A33Z&sr\=c&sp\=w&sig\=uvnsvT654FHOxCII7nhaTzTyW%2BIcXORcUnnQqGeWDtk%3D","expiration"\:1674549993621}
 d0a20b72-1889-4bdd-a2ae-bc14be5c7b34-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2022-12-07T14%3A12%3A21Z&sr\=c&sp\=w&sig\=Os4JIY3EsUv0g7cuqsXRkr%2BPF4gHj6yPqueBNWhet6Q%3D","expiration"\:1670422341018}
 company1assets-key1=key1
-c42df689-100c-4111-b1e0-1af87c09e75a-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2022-12-07T14%3A07%3A18Z&sr\=c&sp\=w&sig\=FQqLkCUDQNtpJziMXtz6QzKlMlMtOCcjQKfF9CpgaBw%3D","expiration"\:1670422038984}

--- a/system-tests/resources/vault/company2/company2-vault.properties
+++ b/system-tests/resources/vault/company2/company2-vault.properties
@@ -1,5 +1,8 @@
-#Wed Dec 07 13:12:21 UTC 2022
+#Wed Jan 25 15:30:41 UTC 2023
 b77fb664-30d7-4351-9c16-22f1ce71d2b0-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2022-12-07T14%3A10%3A32Z&sr\=c&sp\=w&sig\=x8uJEpVcSsk2JmNbjHyTBLe02Dy3aUeB8Y4pagDasZA%3D","expiration"\:1670422232589}
+7719e0bc-48a7-493f-a5a1-2b029252a891-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2023-01-25T16%3A30%3A40Z&sr\=c&sp\=w&sig\=7SsnR4c%2FjdXpl0n2IrHOg1fhyqypTPBlGhayZWnL3ao%3D","expiration"\:1674664240250}
+711c9e7c-6a94-4b6d-8f7a-36cf71cbc5bf-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2023-01-25T16%3A14%3A26Z&sr\=c&sp\=w&sig\=b%2FpbL5OmIx20MRA%2FMI1OtBOTuUu%2FSP4JTSxrr2piSQ0%3D","expiration"\:1674663266839}
+c42df689-100c-4111-b1e0-1af87c09e75a-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2022-12-07T14%3A07%3A18Z&sr\=c&sp\=w&sig\=FQqLkCUDQNtpJziMXtz6QzKlMlMtOCcjQKfF9CpgaBw%3D","expiration"\:1670422038984}
+2011fb1c-8fb8-497a-a687-418c2f2e5996-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2023-01-24T08%3A46%3A33Z&sr\=c&sp\=w&sig\=uvnsvT654FHOxCII7nhaTzTyW%2BIcXORcUnnQqGeWDtk%3D","expiration"\:1674549993621}
 d0a20b72-1889-4bdd-a2ae-bc14be5c7b34-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2022-12-07T14%3A12%3A21Z&sr\=c&sp\=w&sig\=Os4JIY3EsUv0g7cuqsXRkr%2BPF4gHj6yPqueBNWhet6Q%3D","expiration"\:1670422341018}
 company2assets-key1=key2
-c42df689-100c-4111-b1e0-1af87c09e75a-container={"edctype"\:"dataspaceconnector\:azuretoken","sas"\:"?sv\=2021-08-06&se\=2022-12-07T14%3A07%3A18Z&sr\=c&sp\=w&sig\=FQqLkCUDQNtpJziMXtz6QzKlMlMtOCcjQKfF9CpgaBw%3D","expiration"\:1670422038984}

--- a/system-tests/src/test/java/org/eclipse/edc/system/tests/identityhub/IdentityHubIntegrationTest.java
+++ b/system-tests/src/test/java/org/eclipse/edc/system/tests/identityhub/IdentityHubIntegrationTest.java
@@ -14,21 +14,17 @@
 
 package org.eclipse.edc.system.tests.identityhub;
 
-import com.fasterxml.jackson.databind.JsonNode;
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.databind.node.TextNode;
-import okhttp3.OkHttpClient;
+import io.restassured.RestAssured;
 import org.assertj.core.api.AbstractCollectionAssert;
-import org.assertj.core.api.IterableAssert;
 import org.assertj.core.api.ObjectAssert;
 import org.assertj.core.api.ThrowingConsumer;
 import org.eclipse.edc.identityhub.client.IdentityHubClientImpl;
-import org.eclipse.edc.identityhub.credentials.jwt.JwtCredentialEnvelope;
 import org.eclipse.edc.identityhub.credentials.jwt.JwtCredentialEnvelopeTransformer;
 import org.eclipse.edc.identityhub.spi.credentials.model.CredentialEnvelope;
 import org.eclipse.edc.identityhub.spi.credentials.transformer.CredentialEnvelopeTransformerRegistryImpl;
 import org.eclipse.edc.junit.annotations.ComponentTest;
-import org.eclipse.edc.spi.monitor.ConsoleMonitor;
+import org.eclipse.edc.spi.http.EdcHttpClient;
+import org.eclipse.edc.spi.types.TypeManager;
 import org.eclipse.edc.system.tests.utils.TestUtils;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.extension.ExtensionContext;
@@ -37,15 +33,15 @@ import org.junit.jupiter.params.provider.Arguments;
 import org.junit.jupiter.params.provider.ArgumentsProvider;
 import org.junit.jupiter.params.provider.ArgumentsSource;
 
+import java.io.IOException;
 import java.util.Collection;
-import java.util.Map;
 import java.util.stream.Stream;
 
-import static java.util.Optional.ofNullable;
 import static java.util.concurrent.TimeUnit.SECONDS;
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.awaitility.Awaitility.await;
-import static org.eclipse.edc.junit.testfixtures.TestUtils.testOkHttpClient;
+import static org.eclipse.edc.junit.testfixtures.TestUtils.testHttpClient;
+import static org.hamcrest.Matchers.equalTo;
 import static org.junit.jupiter.params.provider.Arguments.arguments;
 
 @ComponentTest
@@ -56,47 +52,44 @@ class IdentityHubIntegrationTest {
     private static final String COMPANY3_IDENTITY_HUB_URL = TestUtils.requiredPropOrEnv("COMPANY3_IDENTITY_HUB_URL", "http://localhost:7173/api/v1/identity/identity-hub");
     private static final String AUTHORITY_IDENTITY_HUB_URL = TestUtils.requiredPropOrEnv("AUTHORITY_IDENTITY_HUB_URL", "http://localhost:7174/api/v1/identity/identity-hub");
 
-    private static final OkHttpClient OK_HTTP_CLIENT = testOkHttpClient();
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
-    private static final ConsoleMonitor CONSOLE_MONITOR = new ConsoleMonitor();
+    private static final EdcHttpClient HTTP_CLIENT = testHttpClient();
+    private static final TypeManager TYPE_MANAGER = new TypeManager();
 
     private IdentityHubClientImpl client;
-
 
     @BeforeEach
     void setUp() {
         var transformerRegistry = new CredentialEnvelopeTransformerRegistryImpl();
-        transformerRegistry.register(new JwtCredentialEnvelopeTransformer(OBJECT_MAPPER));
-        client = new IdentityHubClientImpl(OK_HTTP_CLIENT, OBJECT_MAPPER, CONSOLE_MONITOR, transformerRegistry);
+        transformerRegistry.register(new JwtCredentialEnvelopeTransformer(TYPE_MANAGER.getMapper()));
+        client = new IdentityHubClientImpl(HTTP_CLIENT, TYPE_MANAGER, transformerRegistry);
     }
 
     @ParameterizedTest
     @ArgumentsSource(ParticipantsIdentityHubsUrlProvider.class)
     void retrieveVerifiableCredentials(String hubUrl, String region, String country) {
-        await()
-                .atMost(20, SECONDS)
+        await().atMost(20, SECONDS)
                 .pollInterval(2, SECONDS)
                 .untilAsserted(() -> twoCredentialsInIdentityHub(hubUrl));
 
-        twoCredentialsInIdentityHub(hubUrl)
-                .anySatisfy(vcRequirements("region", region))
-                .anySatisfy(vcRequirements("gaiaXMember", "true"));
+        twoCredentialsInIdentityHub(hubUrl).anySatisfy(vcRequirements("region", region));
     }
 
     @ParameterizedTest
     @ArgumentsSource(DataspaceIdentityHubsUrlProvider.class)
-    void getSelfDescription(String hubUrl, String region, String country) {
-        await().atMost(20, SECONDS)
-                .pollInterval(2, SECONDS)
-                .untilAsserted(() -> selfDescriptionRetrieved(hubUrl));
-
-        selfDescriptionRetrieved(hubUrl).anySatisfy(selfDescriptionRequirements(country));
+    void getSelfDescription(String hubUrl, String region, String country) throws IOException {
+        RestAssured.given()
+                .baseUri(hubUrl)
+                .get("/self-description")
+                .then()
+                .assertThat()
+                .statusCode(200)
+                .body("selfDescriptionCredential.credentialSubject.gx-participant:headquarterAddress.gx-participant:country.@value", equalTo(country));
     }
 
     private static final class ParticipantsIdentityHubsUrlProvider implements ArgumentsProvider {
 
         @Override
-        public Stream<? extends Arguments> provideArguments(ExtensionContext extensionContext) throws Exception {
+        public Stream<? extends Arguments> provideArguments(ExtensionContext extensionContext) {
             return Stream.of(
                     arguments(COMPANY1_IDENTITY_HUB_URL, "eu", "FR"),
                     arguments(COMPANY2_IDENTITY_HUB_URL, "eu", "DE"),
@@ -118,56 +111,25 @@ class IdentityHubIntegrationTest {
         }
     }
 
-    private ThrowingConsumer<JsonNode> selfDescriptionRequirements(String country) {
-        return json -> {
-            var credentialSubject = getOrThrow(json, "credentialSubject");
-            var headquarterAddress = getOrThrow(credentialSubject, "gx-participant:headquarterAddress");
-            var participantCountry = getOrThrow(headquarterAddress, "gx-participant:country");
-            var countryValue = getOrThrow(participantCountry, "@value");
-            assertThat(countryValue).isInstanceOf(TextNode.class);
-            assertThat(countryValue.asText()).isEqualTo(country);
-        };
-    }
-
     private ThrowingConsumer<CredentialEnvelope> vcRequirements(String name, String value) {
         return envelope -> {
-            assertThat(envelope).isInstanceOf(JwtCredentialEnvelope.class);
-            var jwt = ((JwtCredentialEnvelope) envelope).getJwtVerifiableCredentials();
-            var claims = jwt.getJWTClaimsSet();
-            assertThat(claims.getIssuer()).as("Issuer is a Web DID").startsWith("did:web:");
-            assertThat(claims.getSubject()).as("Subject is a Web DID").startsWith("did:web:");
-            assertThat(claims.getClaim("vc")).as("VC")
-                    .isInstanceOfSatisfying(Map.class, t -> {
-                        assertThat(t.get("id"))
-                                .as("VC ID")
-                                .isInstanceOfSatisfying(String.class, s -> assertThat(s).isNotBlank());
-
-                        assertThat(t.get("credentialSubject"))
-                                .as("VC credentialSubject")
-                                .isInstanceOfSatisfying(Map.class, s -> assertThat(s.get(name))
-                                        .as(name)
-                                        .isInstanceOfSatisfying(String.class,
-                                                r -> assertThat(r).isEqualTo(value)));
+            var vcResult = envelope.toVerifiableCredential(TYPE_MANAGER.getMapper());
+            assertThat(vcResult.succeeded()).isTrue();
+            var credential = vcResult.getContent().getItem();
+            assertThat(credential.getIssuer()).as("Issuer is a Web DID").startsWith("did:web:");
+            assertThat(credential.getCredentialSubject().getId()).as("Subject is a Web DID").startsWith("did:web:");
+            assertThat(credential.getCredentialSubject().getClaims())
+                    .extractingByKey(name)
+                    .satisfies(o -> {
+                        assertThat(o).isInstanceOf(String.class);
+                        assertThat((String) o).isEqualTo(value);
                     });
         };
-    }
-
-    private static JsonNode getOrThrow(JsonNode node, String key) {
-        var value = node.get(key);
-        assertThat(value).as("Get value for key: %s", key).isNotNull();
-        return value;
     }
 
     private AbstractCollectionAssert<?, Collection<? extends CredentialEnvelope>, CredentialEnvelope, ObjectAssert<CredentialEnvelope>> twoCredentialsInIdentityHub(String hubUrl) {
         var vcs = client.getVerifiableCredentials(hubUrl);
         assertThat(vcs.succeeded()).isTrue();
-        return assertThat(vcs.getContent()).hasSize(3);
-    }
-
-    private IterableAssert<JsonNode> selfDescriptionRetrieved(String hubUrl) {
-        var vcs = client.getSelfDescription(hubUrl);
-
-        assertThat(vcs.succeeded()).withFailMessage(ofNullable(vcs.getFailureDetail()).orElse("")).isTrue();
-        return assertThat(vcs.getContent());
+        return assertThat(vcs.getContent()).hasSize(2);
     }
 }

